gene2phenotype_project/gene2phenotype_app/serializers/locus_genotype_disease.py

```diff
@@ -447,6 +452,138 @@ def update(self, instance, validated_data):
 
         return instance
 
+    def update_mechanism(self, lgd_instance, validated_data):
+        """
+            Method to update the molecular mechanism of the LGD record.
+            It only allows to update mechanisms with value 'undetermined'
+            and evidence 'inferred'.
+
+            Mandatory fields are molecular_mechanism name and support (inferred/evidence).
+
+            Example:    "molecular_mechanism": {
+                            "name": "gain of function",
+                            "support": "evidence"
+                        },
+                        "mechanism_synopsis": {
+                            "name": "",
+                            "support": ""
+                        },
+                        "mechanism_evidence": [{'pmid': '25099252', 'description': 'text', 'evidence_types': 
+                            [{'primary_type': 'Rescue', 'secondary_type': ['Human', 'Patient Cells']}]}]
+
+        """
+        molecular_mechanism_value = validated_data.get("molecular_mechanism")["name"]
+        molecular_mechanism_support = validated_data.get("molecular_mechanism")["support"]
+        mechanism_synopsis = validated_data.get("mechanism_synopsis", None) # synopsis is optional
+        mechanism_evidence = validated_data.get("mechanism_evidence", None) # evidence is optional
```

Comment:
```suggestion
        mechanism_evidence = validated_data.get("mechanism_evidence", None) # synopsis evidence is optional
```

Reply:
`mechanism_evidence` is the molecular mechanism evidence. The synopsis has no evidence directly linked to it.

Reply:
I'm going to improve these comments.